src/main/scala/tech/cryptonomic/conseil/generic/chain/RemoteRpc.scala

```diff
@@ -0,0 +1,78 @@
+package tech.cryptonomic.conseil.generic.chain
+
+trait RemoteRpc[Eff[_], Req[_], Res[_]] {
+
+  type CallConfig
+  type PostPayload
+
+  def runGetCall[CallId](callConfig: CallConfig, request: Req[CallId], commandMap: CallId => String): Eff[Res[CallId]]
+
+  def runPostCall[CallId](callConfig: CallConfig, request: Req[CallId], commandMap: CallId => String, payload: Option[PostPayload] = None): Eff[Res[CallId]]
+
+}
+
+object RemoteRpc {
+
+  import cats.{Id, Functor}
+  import cats.data.Const
+  import cats.syntax.functor._
+
+  type Aux[Eff[_], Req[_], Res[_], Conf, Payload] = RemoteRpc[Eff, Req, Res] { type CallConfig = Conf; type PostPayload = Payload }
+
+  def apply[Eff[_], Req[_], Res[_], CallConfig, Payload](
+    implicit remote: Aux[Eff, Req, Res, CallConfig, Payload]
+  ): Aux[Eff, Req, Res, CallConfig, Payload] = remote
+
+  def runGet[Eff[_], CallId, Req[_], Res[_], CallConfig](
+    callConfig: CallConfig,
+    request: Req[CallId],
+    commandMap: CallId => String
+  )(implicit remote: Aux[Eff, Req, Res, CallConfig, _]): Eff[Res[CallId]] =
+    remote.runGetCall(callConfig, request, commandMap)
+
+  def runGet[Eff[_], CallId, Req[_], Res[_]](
+    request: Req[CallId],
+    commandMap: CallId => String
+  )(implicit remote: Aux[Eff, Req, Res, Any, _]): Eff[Res[CallId]] =
+    remote.runGetCall((), request, commandMap)
+
+  def runGet[Eff[_]: Functor, CallId, Res](
+    request: CallId,
+    commandMap: CallId => String
+  )(implicit remote: Aux[Eff, Id, Const[Res, ?], Any, _]): Eff[Res] =
+    remote.runGetCall((), request, commandMap).map(_.getConst)
```

Comment:
For my education, what does `getConst()` do?

Reply:
`Const[A, B]` is the type analogous to the function 
```scala
//a function taking two params that always returns the first
// it's some sort of hi-jacked function, you give it the first value and you get back a 
// another function that returns such value, independently of the passed argument 
def const[A, B](a: A)(b: B) = a
```
Now to the Const type, and its `getConst` attribute:
I had need to define the Rpc call to be able to return a value that might be tied to the input arguments (e.g. a tuple of the input/output), hence the parametric `Res[_]` return type, that can be used to return a `Res[CallId]` when multiple (lists) inputs (of type CallId) are passed as a batch.
In the case where we want the output to actually be independent of the `CallId` type, we can use the type
`Const[A, B]` which ignores the `B` type and is actually an alias for `A`
For the `runGet` that wants to return a simple result for a single request, the type `Const[Res, ?]` is a type constructor that takes one type, as expected by the generic form of RemoteRpc.Aux, but ignores it.

tl;dr; It all boils down to having a valid way to return a `Eff[Res]` that is independent of the single request type input `CallId`.
Even more, the evolution of this code is a huge simplification of all this, I found this encoding too general and ambitious, so I made it much simpler, at the cost of not being able to accomodate multiple designs in a single "form".
I'll try to push a PR out for you all to evaluate as soon as I can, based on my spare time